src/ralph/backends/data/fs.py

```diff
@@ -0,0 +1,274 @@
+    ) -> int:
+        """Write stream records to the target file and return their count.
+
+        Args:
+            stream: (Iterable): The stream containing data to write.
+            target (str or None): The target file path.
+                If target is a relative path, it is considered to be relative to the
+                    `default_directory_path`.
+                If target is `None`, a random (uuid4) file is created in the
+                    `default_directory_path` and used as the target instead.
+            chunk_size (int or None): Ignored. Is equal to the stream chunk size.
+            ignore_errors (bool): Ignored.
+            operation_type (BaseOperationType or None): The mode of the write operation.
+                If operation_type is `CREATE` or `INDEX`, the target file is expected to
+                    be absent. If the target file exists a `FileExistsError` is raised.
+                If operation_type is `UPDATE`, the target file is overwritten.
+                If operation_type is `APPEND`, the stream content is appended to the
+                    end of the target file.
+
+        Returns:
+            int: The number of written files. (Always equal to 1).
+        """
+
+        if not operation_type:
+            operation_type = self.default_operation_type
+
+        if operation_type == BaseOperationType.DELETE:
+            msg = "Delete operation_type is not allowed."
+            logger.error(msg)
+            raise BackendParameterException(msg)
+
+        if not target:
+            target = str(uuid4())
+            logger.info("Target file not specified; using random file name: %s", target)
+
+        target = Path(target)
+        path = target if target.is_absolute() else self.default_directory / target
+
+        if operation_type in [BaseOperationType.CREATE, BaseOperationType.INDEX]:
+            if path.is_file():
+                msg = (
+                    "%s already exists and overwrite is not allowed with operation_type"
+                    " create or index."
+                )
+                logger.error(msg, path)
+                raise FileExistsError(msg % path)
+
+            logger.debug("Creating file: %s", path)
+
+        mode = "wb"
+        if operation_type in [BaseOperationType.APPEND]:
+            mode = "ab"
+            logger.debug("Appending to file: %s", path)
+
+        with path.open(mode) as file:
+            for chunk in stream:
+                file.write(chunk)
+
+        # The file has been created, add a new entry to the history.
+        self.append_to_history(
+            {
+                "backend": self.name,
+                "command": "push",
+                "id": str(path),
+                "filename": path.name,
+                "size": path.stat().st_size,
+                "pushed_at": now(),
+            }
+        )
+        return 1
+
+    @staticmethod
+    def _read_raw(file, chunk_size):
```

Comment:
```suggestion
    def _read_raw(file: IO, chunk_size: int) -> Generator[bytes, None, None]:
```

Reply:
Well spotted) Thanks) updated this part)